File: Instructor Andres Moreno/formularios/14. salario/js/salario.js
/* 
fecha 17/06/2024
funcion salario de una persona (Nomina)
autor: Santiago GM
*/

function datos() {

  let diasT = parseInt(document.getElementById('diaT').value)
  let valorD = parseFloat(document.getElementById('valorD').value.replace(/[,\.]/g, ''))
  let pantalla = document.getElementById('pantalla')

  if (diasT && valorD ) {
    
    pantalla.innerHTML = 
    `
      <h1>Salario</h1>
      <p>
          Su salario Bruto es de: $${sueldo(diasT,valorD)} <br>
          El porcenta de Salud es de: ${salud(sueldo(diasT,valorD))} <br>
          El porcenta de  Pensión es de: ${pension(sueldo(diasT,valorD))} <br>
          El porcenta de ARL es de: ${arl(sueldo(diasT,valorD))} <br>
          El subsidio de trasporte es de: $${subTras(sueldo(diasT,valorD))} <br>
          La retencion es de:  ${reten(sueldo(diasT,valorD)) }<br>
          Su salario Neto será de: ${pagoT(sueldo(diasT,valorD))}
      </p>
    `
  } else {
    pantalla.innerHTML = `<strong>Todos los campos deben estar llenos </strong>`
    
  }


  return false
}

function sueldo(diaT, valorD) {
  let pago = diaT * valorD;
  return pago;
}
function salud(pago) {
  let saludP = pago * 0.12;
  return saludP;
}

function pension(pago) {
  let pensionP = pago * 0.16;
  return pensionP;
}

function arl(pago) {
  let arlP = pago * 0.052;
  return arlP;
}
function subTras(pago) {
  let salarioM = 1300000;
  let trans;
  if (pago <= salarioM * 2) {
    trans = 114000;
  } else {
    trans = 0;
  }
  return trans;
}
function reten(pago) {
  let salarioM = 1300000;
  let retencion;
  if (pago > salarioM * 4) {
    retencion = pago * 0.04;
  } else {
    retencion = 0;
  }
  return retencion;
}


function pagoT(pago) {
  
  let saludE = salud(pago);
  let pensionE = pension(pago);
  let arlE = arl(pago);
  let retencion = reten(pago);
  let subTrasporte = subTras(pago);

  let descuento = saludE + pensionE + arlE;

  let totalSalario = pago + subTrasporte - (retencion + descuento);

  return totalSalario;
}

File: python/14. salario/utils/salario.py
#Calcular el sueldo de una persona
#Autor: Santiago GM


# Como paramétro

def sueldo(dias_trabajados_p, valor_dia_p) :
  dias_trabajados, valor_dia = dias_trabajados_p, valor_dia_p
  pago = dias_trabajados * valor_dia
  return pago

def salud(pago) :
  salud_p = pago * 0.12
  return salud_p

def pension(pago) :
  pension_p = pago * 0.16
  return pension_p

def arl(pago) :
  arl_p = pago * 0.052
  return arl_p

def subsidio_transprote(pago) :
  salarioM = 1300000
  if (pago <= salarioM * 2) :
    trans = 114000
  else :
    trans = 0
  return trans

def reten(pago) :
  salarioM = 1300000
  if (pago > salarioM * 4) :
    retencion = pago * 0.04
  else :
    retencion = 0

  return retencion

def pago_total(pago) :
  salario = pago
  salud_parametro = salud(pago)
  pension_parametro = pension(pago)
  arl_parametro = arl(pago)

  retencion = reten(pago)
  subTrasporte = subsidio_transprote(pago)
  descuento = salud_parametro + pension_parametro + arl_parametro

  totalSalario = salario + subTrasporte - (retencion + descuento)

  return totalSalario


#función lambda
sueldo_lamb = lambda dias_trabajados_p, valor_dia_p : dias_trabajados_p * valor_dia_p

salud_lamb = lambda pago :  pago * 0.12

pension_lamb = lambda pago : pago * 0.16

arl_lamb = lambda pago : pago * 0.052

subsido_transporte_lamb = lambda pago : 114000 if (pago <= 1300000 * 2) else 0

reten_lamb = lambda pago : 0 if (pago <= 1300000 * 4) else (pago * 0.04) 

pago_total_lamb = lambda pago : pago + subsido_transporte_lamb(pago) - (reten_lamb(pago) + salud_lamb(pago) + pension_lamb(pago) + arl_lamb(pago))
